### PR TITLE
New Slack Formatting Cheat Sheet

### PR DESCRIPTION
**Pull request by whitesunset**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.

- [x] New Instant Answer
    - [x] Cheat Sheets: **`New Slack Formatting Cheat Sheet`**
    - [ ] Other: **`New {IA Name} Instant Answer`**
- [ ] Improvement
    - [ ] Bug fix: **`{IA Name}: Fix {Issue number or one-line description}`**
    - [ ] Enhancement: **`{IA Name}: {Description of Improvements}`**
- [ ] Non–Instant Answer
    - [ ] Other (Role, Template, Test, Documentation, etc.): **`{GoodieRole/Templates/Tests/Docs}: {Short Description}`**

###### Description of changes

Markdown message formatting for Slack

###### Which issues (if any) does this fix?

Hello!

I tried to pass tests via `duckpan test CheatSheets` and every time got same error `Failed test 'FAIL: No Instant Answer page found with ID 'slack_formatting_cheat_sheet'' at t/CheatSheets/CheatSheetsJSON.t line 274.` even after I create IA page with proper ID. 

What I missed? Visual test passed, cheat sheet triggers correctly and looks as I expected. 

###### People to notify (@mention interested parties)

---

Instant Answer Page: https://duck.co/ia/view/slack_formatting_cheat_sheet

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @mention

### CODE REVIEW

**Comment by daxtheduck** [comment]
@whitesunset Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please head to https://quackslack.herokuapp.com/ to get an invite.

>More Info: http://docs.duckduckhack.com/submitting/submitting-overview.html

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by edgesince84** [comment]
This should be useful!  thanks again @whitesunset

File: share/goodie/cheat_sheets/json/slack-formatting.json [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  "description": "Slack is a team collaboration tool",
+  "metadata": {
+    "sourceName": "Slack",
+    "sourceUrl": "https://get.slack.help/hc/en-us/articles/202288908-Formatting-your-messages"
+  },
+  "aliases": [
+    "slack markdown",
+    "slack tags",
+    "slack message",      
+    "slack messages"
```
Comment (ManrajGrover) [inline share/goodie/cheat_sheets/json/slack-formatting.json]: You can add more alias like `slack text format`, `slack code blocks`, `slack text blockquotes`. :blush:

Comment (whitesunset) [inline share/goodie/cheat_sheets/json/slack-formatting.json]: Thanks for suggestion, just added.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Slack Formatting Cheatsheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/slack_formatting_cheat_sheet_mobile.png?rIXCYBMzcz)

File: share/goodie/cheat_sheets/json/slack-formatting.json [hunk 2]
```diff
@@ -0,0 +1,74 @@
+{
+  "id": "slack_formatting_cheat_sheet",
+  "name": "Slack",
+  "description": "Slack is a team collaboration tool",
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/slack-formatting.json]: Maybe change this to something like "Message formatting in Slack, the team collaboration tool"

File: share/goodie/cheat_sheets/json/slack-formatting.json [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  "aliases": [
+    "slack markdown",
+    "slack tags",
+    "slack message",      
+    "slack messages",
+    "slack text format",
+    "slack code blocks",
+    "slack text blockquotes"
+  ],
+  "template_type": "keyboard",
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/slack-formatting.json]: Please could you change the template type to "terminal", which gives more space for long lines? See https://duckduckgo.com/?q=markdown+cheatsheet for an example.

File: share/goodie/cheat_sheets/json/slack-formatting.json [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    "sourceUrl": "https://get.slack.help/hc/en-us/articles/202288908-Formatting-your-messages"
+  },
+  "aliases": [
+    "slack markdown",
+    "slack tags",
+    "slack message",      
+    "slack messages",
+    "slack text format",
+    "slack code blocks",
+    "slack text blockquotes"
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/slack-formatting.json]: Maybe add things like "slack format" and "slack text formatting" here.

Comment (whitesunset) [inline share/goodie/cheat_sheets/json/slack-formatting.json]: @tagawa thanks! done

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Slack Formatting Cheatsheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/slack_formatting_cheat_sheet_mobile.png?e1XIRMUtxU)

**Comment by zekiel** [comment]
Thanks @whitesunset! This looks good to merge and will be useful for [our slack team](https://quackslack.herokuapp.com/)!

**Comment by daxtheduck** [comment]
@whitesunset this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

**Comment by whitesunset** [comment]
@zekiel thank you very much!

**Comment by edgesince84** [comment]
Awesome, thanks again @whitesunset 🎉

**Pull request merged**